x/gov/abci.go

```diff
@@ -49,12 +49,20 @@ func EndBlocker(ctx sdk.Context, keeper keeper.Keeper) {
 
 		passes, burnDeposits, tallyResults := keeper.Tally(ctx, proposal)
 
-		if burnDeposits {
-			keeper.DeleteDeposits(ctx, proposal.ProposalId)
-		} else {
-			keeper.RefundDeposits(ctx, proposal.ProposalId)
+		// If an expedited proposal fails, we do not want to update
+		// the deposit at this point since the proposal is converted to regular.
+		// As a result, the deposits are either deleted or refunded in all casses
+		// EXCEPT when an expedited proposal fails.
+		if !(proposal.IsExpedited && !passes) {
+			if burnDeposits {
+				keeper.DeleteDeposits(ctx, proposal.ProposalId)
+			} else {
+				keeper.RefundDeposits(ctx, proposal.ProposalId)
+			}
 		}
 
+		keeper.RemoveFromActiveProposalQueue(ctx, proposal.ProposalId, proposal.VotingEndTime)
```

Comment:
should this logic be here? won't we end up removing active proposals from the queue as well?

Reply:
+1

Reply:
The original logic is to remove it anyway. I think we move it here because we will have to add it back later if this is an expedited proposal, then it would become a regular one.

Reply:
It was move from line 93 to here. The reason is because we need to add it back on line 104